src/Featurizers/Components/StatisticsMetricsEstimator.h

```diff
@@ -0,0 +1,391 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "../../Traits.h"
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         StatisticalMetricsEstimatorName("StatisticalMetricsEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         BasicStatisticalAnnotationData
+///  \brief         An annotation class which contains the min, max and count
+///                 for an input column. Template T can be all types
+///
+template <typename T>
+class BasicStatisticalAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    T             const                           Min;
+    T             const                           Max;
+    std::uint64_t  const                          Count;

```

Comment:
Can you remove one of the spaces after uint64_t since you went to the work to horizontally align the values.